engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1777,16 +1644,52 @@ IR.Comment translateComment(Tree tree) {
   }
 
   private Option<IdentifiedLocation> getIdentifiedLocation(Tree ast) {
-    if (ast == null) {
-      return Option.empty();
+    return Option.apply(ast).map(ast_ -> {
+      int begin = Math.toIntExact(ast_.getStartCode());
+      int end = Math.toIntExact(ast_.getEndCode());
+      return new IdentifiedLocation(new Location(begin, end), Option.empty());
+    });
+  }
+
+  private Option<IdentifiedLocation> getIdentifiedLocation(ArgumentDefinition ast) {
+    // Note: In the IR, `DefinitionArgument` is a type of AST node; in the parser, it is not an AST-level type, but a
+    // type used only in specific locations. As a result, the IR expects it to have a source-code location, but the
+    // parser doesn't have a span for it. Here we synthesize one. This will not be necessary if we refactor IR so that
+    // `ArgumentDefinition` is not an AST node, though that change would have effects throughout the compiler and may
+    // not be worthwhile if IR is expected to be replaced.
+    long begin;
+    if (ast.getOpen() != null) {
+      begin = ast.getOpen().getStartCode();
+    } else if (ast.getOpen2() != null) {
+      begin = ast.getOpen2().getStartCode();
+    } else if (ast.getSuspension() != null) {
+      begin = ast.getSuspension().getStartCode();
```

Comment:
Open, open2, suspension!? That's complicated. Simple `ArgumentDefinition.getOpen()` with proper behavior would be better.

Reply:
`getOpen` returns the outer parenthesis, if present. This logic is finding the textual begin and end of the `ArgumentDefinition`; which token begins the object depends on what syntax occurred within the argument definition. A pattern is always present, but there are optional parts that can extend to either side of it: A type annotation, a default value, and a `~` suspension operator, along with up to two pairs of parentheses (one around the type annotation and one around the default value). As noted in the comment earlier in the function, I think the proper solution is to refactor the IR so that its `DefinitionArgument` is not a node, and doesn't need a code span; however, I looked in to doing that, and it would require changes all across the compiler. I don't think it's worth it; this works.